Intermittent test failure: WebExtensionSupportTest: "reacts to browser action popup being toggled by opening and closing tab"

```
[task 2020-01-09T11:21:19.454Z]   TEST: reacts to browser action popup being toggled by opening and closing tab
[task 2020-01-09T11:21:19.454Z]     [Robolectric] mozilla.components.support.webextensions.WebExtensionSupportTest.reacts to browser action popup being toggled by opening and closing tab: sdk=28; resources=binary
[task 2020-01-09T11:21:19.454Z]     I/MonitoringInstr: Instrumentation started!
[task 2020-01-09T11:21:19.455Z]     I/MonitoringInstr: Setting context classloader to 'org.robolectric.internal.bytecode.SandboxClassLoader@5fbd2deb', Original: 'org.robolectric.internal.bytecode.SandboxClassLoader@5fbd2deb'
[task 2020-01-09T11:21:19.455Z]     java.lang.instrument.IllegalClassFormatException: Error while instrumenting mozilla/components/browser/state/reducer/WebExtensionReducer$reduce$4.
[task 2020-01-09T11:21:19.455Z]     at org.jacoco.agent.rt.internal_43f5073.CoverageTransformer.transform(CoverageTransformer.java:94)
[task 2020-01-09T11:21:19.455Z]     at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
[task 2020-01-09T11:21:19.455Z]     at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
[task 2020-01-09T11:21:19.455Z]     at java.lang.ClassLoader.defineClass1(Native Method)
[task 2020-01-09T11:21:19.455Z]     at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
[task 2020-01-09T11:21:19.455Z]     at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
[task 2020-01-09T11:21:19.455Z]     at org.robolectric.internal.bytecode.SandboxClassLoader.maybeInstrumentClass(SandboxClassLoader.java:134)
[task 2020-01-09T11:21:19.455Z]     at org.robolectric.internal.bytecode.SandboxClassLoader.lambda$findClass$0(SandboxClassLoader.java:111)
[task 2020-01-09T11:21:19.455Z]     at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:50)
[task 2020-01-09T11:21:19.455Z]     at org.robolectric.internal.bytecode.SandboxClassLoader.findClass(SandboxClassLoader.java:110)
[task 2020-01-09T11:21:19.455Z]     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[task 2020-01-09T11:21:19.455Z]     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[task 2020-01-09T11:21:19.455Z]     at mozilla.components.browser.state.reducer.WebExtensionReducer.reduce(WebExtensionReducer.kt:46)
[task 2020-01-09T11:21:19.456Z]     at mozilla.components.browser.state.reducer.BrowserStateReducer.reduce(BrowserStateReducer.kt:36)
[task 2020-01-09T11:21:19.456Z]     at mozilla.components.browser.state.store.BrowserStore$1.invoke(BrowserStore.kt:24)
[task 2020-01-09T11:21:19.456Z]     at mozilla.components.browser.state.store.BrowserStore$1.invoke(BrowserStore.kt:20)
[task 2020-01-09T11:21:19.456Z]     at mozilla.components.lib.state.Store.dispatchInternal(Store.kt:102)
[task 2020-01-09T11:21:19.456Z]     at mozilla.components.lib.state.Store.access$dispatchInternal(Store.kt:40)
[task 2020-01-09T11:21:19.456Z]     at mozilla.components.lib.state.Store$dispatch$1.invokeSuspend(Store.kt:97)
[task 2020-01-09T11:21:19.456Z]     at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[task 2020-01-09T11:21:19.456Z]     at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
[task 2020-01-09T11:21:19.457Z]     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[task 2020-01-09T11:21:19.457Z]     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[task 2020-01-09T11:21:19.457Z]     at java.lang.Thread.run(Thread.java:748)
[task 2020-01-09T11:21:19.457Z]     Caused by: java.io.IOException: Error while instrumenting 
mozilla/components/browser/state/reducer/WebExtensionReducer$reduce$4.
[task 2020-01-09T11:21:19.457Z]     at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrumentError(Instrumenter.java:159)
[task 2020-01-09T11:21:19.457Z]     at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrument(Instrumenter.java:109)
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.CoverageTransformer.transform(CoverageTransformer.java:92)
[task 2020-01-09T11:21:19.462Z]     ... 23 more
[task 2020-01-09T11:21:19.462Z]     Caused by: java.lang.IllegalStateException: Cannot process instrumented class mozilla/components/browser/state/reducer/WebExtensionReducer$reduce$4. Please supply original non-instrumented classes.
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.core.internal.instr.InstrSupport.assertNotInstrumented(InstrSupport.java:238)
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.core.internal.instr.ClassInstrumenter.visitField(ClassInstrumenter.java:56)
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.asm.ClassVisitor.visitField(ClassVisitor.java:287)
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.asm.ClassReader.readField(ClassReader.java:907)
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.asm.ClassReader.accept(ClassReader.java:683)
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.asm.ClassReader.accept(ClassReader.java:400)
[task 2020-01-09T11:21:19.462Z]     at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrument(Instrumenter.java:89)
[task 2020-01-09T11:21:19.463Z]     at org.jacoco.agent.rt.internal_43f5073.core.instr.Instrumenter.instrument(Instrumenter.java:107)
[task 2020-01-09T11:21:19.463Z]     ... 24 more
[task 2020-01-09T11:21:19.548Z]   FAILURE
```

https://firefox-ci-tc.services.mozilla.com/tasks/BnsmZc9qR568Bb7tZgHCeg/runs/0/logs/https%3A%2F%2Ffirefox-ci-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FBnsmZc9qR568Bb7tZgHCeg%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log

--- a/components/support/webextensions/build.gradle
+++ b/components/support/webextensions/build.gradle
@@ -42,6 +42,7 @@ dependencies {
     implementation Dependencies.kotlin_coroutines
 
     testImplementation project(':support-test')
+    testImplementation project(':support-test-libstate')
 
     testImplementation Dependencies.androidx_test_core
     testImplementation Dependencies.androidx_test_junit

--- a/components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt
+++ b/components/support/webextensions/src/test/java/mozilla/components/support/webextensions/WebExtensionSupportTest.kt
@@ -26,6 +26,7 @@ import mozilla.components.support.test.any
 import mozilla.components.support.test.argumentCaptor
 import mozilla.components.support.test.eq
 import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.libstate.ext.waitUntilIdle
 import mozilla.components.support.test.mock
 import mozilla.components.support.test.whenever
 import org.junit.After
@@ -374,6 +375,8 @@ class WebExtensionSupportTest {
         actionCaptor = argumentCaptor()
         verify(store, times(6)).dispatch(actionCaptor.capture())
         assertEquals(popupSessionId, (actionCaptor.value as TabListAction.SelectTabAction).tabId)
+
+        store.waitUntilIdle()
 
         // Toggling again should close tab
         delegateCaptor.value.onToggleBrowserActionPopup(ext, engineSession, browserAction)

